src/SDKKeysCache.ts

```diff
@@ -1,20 +1,34 @@
 import { SDKKeysCacheInterface } from "./interfaces/SDKKeyCacheInterface";
 
 export default class SDKKeysCache implements SDKKeysCacheInterface {
-  private cache: Set<string> | null;
+  private cache: {
+    allKeys: Set<string> | null;
+    globalKeys: Set<string> | null;
+  };
   public constructor() {
-    this.cache = null;
+    this.cache = { allKeys: null, globalKeys: null };
   }
 
   get(): Promise<Set<string> | null> {
-    return Promise.resolve(this.cache);
+    return Promise.resolve(this.cache.allKeys);
   }
   set(keys: Set<string>): Promise<void> {
-    this.cache = keys;
+    this.cache.allKeys = keys;
     return Promise.resolve();
   }
   clear(): Promise<void> {
```

Comment:
nit: adding a `clearAll` here could keep the logic of clearing global and all keys contained to the sdk key cache and you wouldn't need to handle that in the cache utils later.

Reply:
Yeah makes sense. I just didn't want to make a breaking change to the interface just incase that makes it harder to implement the changes